features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapReceiverImpl.java

```diff
@@ -106,7 +106,11 @@ public void setTrapdConfig(TrapdConfiguration newTrapdConfig) {
      */
     public static boolean compareSnmpV3UsersMap(Map<String, SnmpV3User> existingSnmpV3UserMap, Map<String, SnmpV3User> updatedSnmpV3Usermap) {
 
-        if (existingSnmpV3UserMap.isEmpty() || updatedSnmpV3Usermap.isEmpty()) {
+        if ((existingSnmpV3UserMap.isEmpty() && updatedSnmpV3Usermap.isEmpty())
+                || (!existingSnmpV3UserMap.isEmpty() && updatedSnmpV3Usermap.isEmpty())) {
```

Comment:
This is intended because earlier this was the step which i missed,ie when both list  are empty,still it was considering that some data is there to get updated and hence it was restarting everytime.Now once this check is done we are ensuring that for empty case also we dont need to check anything and this works well.